### PR TITLE
Fix false positives for interpolation in property name in custom-property-pattern

### LINKED ISSUE #5945: Fix false positives for interpolation in property name in custom-property-pattern
### What steps are needed to reproduce the bug?

Have this scss

```scss
.a {
    --#{$variable-prefix}table-bg: rgba(#a1f2b0, 0.3);
}
```

### What Stylelint configuration is needed to reproduce the bug?

`stylelint-config-standard`

### How did you run Stylelint?

CLI with `stylelint "**/*.css" --config myconfig.json`

### Which version of Stylelint are you using?

`14.0.0`

### What did you expect to happen?

No errors

### What actually happened?

Triggered `custom-property-pattern rule` error.

### Does the bug relate to non-standard syntax?

scss

### Proposal to fix the bug

Allow variables syntax in `custom-property-pattern` rule

(https://github.com/stylelint/stylelint-config-standard/issues/240)

### PR DESCRIPTION
**Pull request by kawaguchi1102**
> Which issue, if any, is this issue related to?

Closes #5945

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/custom-property-pattern/__tests__/index.js [hunk 1]
```diff
@@ -138,5 +138,14 @@ testRule({
 		{
 			code: 'a { color: var(tokens.$bar-color); }',
 		},
+		{
+			code: 'a { --#{$variable-prefix}table-bg: rgba(#a1f2b0, 0.3); }',
+		},
+		{
+			code: 'a { color: var( #{$variable-color} ); }',
```
Comment (jeddy3) [inline lib/rules/custom-property-pattern/__tests__/index.js:145]:
```suggestion
			code: 'a { color: var( #{$bar} ); }',
```

**Review by jeddy3 — CHANGES_REQUESTED** [review]
@kawaguchi1102 Thanks for jumping on this.

Changes to rule code LGTM.

I've requested changes to the tests (as committable suggestions) to bring them in line with [our conventions](https://stylelint.io/developer-guide/rules#write-tests).

---

Note: You can either refactor in this pull request (or we can create a follow-up issue) to move the duplicated code into `check(property) {}` function. It's up to you.

**Comment by kawaguchi1102** [comment]
@jeddy3 Sorry, I must have missed it.
i've updated it now with what you said.

---

> Note: You can either refactor in this pull request (or we can create a follow-up issue) to move the duplicated code into `check(property) {}` function. It's up to you.

That's a good idea!
I'll refactor with this PR.

**Comment by kawaguchi1102** [comment]
@jeddy3 The refactor has been completed.
I hope I am not mistaken in this refactor.

**Review by jeddy3 — COMMENTED** [review]
Thanks for making the changes. LGTM!

> I hope I am not mistaken in this refactor.

It's looks spot on to me, thank you.

**Review by jeddy3 — APPROVED** [review]

**Review by ybiquitous — APPROVED** [review]
Thank you! LGTM 👍🏼

**Comment by jeddy3** [comment]
> - Fixed: `custom-property-pattern` false positives for interpolation in property name ([#5949](https://github.com/stylelint/stylelint/pull/5949)).

**Pull request merged**